opds_import.py

```diff
@@ -231,48 +240,111 @@ def __init__(self, _db, data_source_name=DataSource.METADATA_WRANGLER,
         self.mirror = mirror
         self.http_get = http_get
 
+
     def import_from_feed(self, feed, even_if_no_author=False, 
                          cutoff_date=None, 
                          immediately_presentation_ready=False):
-        metadata_objs, messages, next_links = self.extract_metadata(feed)
-        imported = []
-        for metadata in metadata_objs:
+
+        imported_editions = {}
+        imported_pools = {}
+        imported_works = {}
+        # status_messages notes business logic errors and non-success download statuses
+        status_messages = {}
+        next_links = []
+        try:
+            metadata_objs, circulation_objs, status_messages, next_links = self.extract_feed_data(feed)
+        except Exception, e:
+            message = StatusMessage(500, "Local exception during import:\n%s" % traceback.format_exc())
+            return imported_editions.values(), imported_pools.values(), imported_works.values(), status_messages, next_links
```

Comment:
there are some problems with this, but I'll take care of it in the next branch since I'll need to change a lot of this anyway.